### PR TITLE
Render pinned posts in compact mode if already read.

### LINKED ISSUE #930: Render pinned posts compact/collapsed/consolidated/etc. after interaction
## Introduction
Briefly: 
- Some communities have many pinned posts (c/mlemapp has 4!). 
- On mobile, users have to scroll past these before interacting with all other posts, even if user has already previously viewed these pinned posts.
- This issue tracks any related discussions relevant to pinned posts UI/UX.
- See branches `/bosco/pinned-posts-ui`, `/bosco/pinned-posts-ui-1` for example implementations.

## Requirements
- [ ] How to make pinned posts more compact, etc.?
- [ ] When to make pinned posts more compact, etc.?
- [ ] Extend post tracker (or the upcoming replacement) to return pinned posts separately.

## Discussion
Reddit introduced a consolidated pinned posts UI on mobile about a year ago to what appears to be a generally negative response from the community (especially from mods):
> https://www.reddit.com/r/ModSupport/comments/ydb66d/consolidated_pinned_posts_on_mobile_is_really/
> https://www.reddit.com/r/redditmobile/comments/15of9h2/ios_2023310_collapsed_pinned_posts_is_ruining/
> https://www.reddit.com/r/redditmobile/comments/15i53bn/ios2023300_pinned_posts_feature_for_subreddits_is/
> https://www.reddit.com/r/modnews/comments/xuvkmy/announcing_consolidated_pinned_posts_on_android/
> https://www.reddit.com/r/ModSupport/comments/xw2jm9/the_new_update_to_pinned_posts_is_already_causing/

Reddit's implementation hides all pinned posts inside a "disclosure group" after two interactions with a community (this may have changed since the original announcement).

Our implementation probably should avoid hiding pinned posts in a disclosure group section, and also allow users to opt-in to auto-collapse/compact etc.

### PR DESCRIPTION
**Pull request by boscojwho**
# Checklist
- [x] I have read CONTRIBUTING.md
- [x] I have described what this PR contains
- [x] If this PR alters the UI, I have attached pictures/videos
- [x] This PR addresses one or more open issues that were assigned to me:
      - closes #930 

# Pull Request Information

## About this Pull Request
Render pinned posts (`FeedPost`) in compact mode if already read.

### CODE REVIEW

**Comment by EricBAndrews** [comment]
What do we think about enabling this only within a community? It works really nicely when the pinned posts are all collected at the top but it's a little weird to come across a random compact post when scrolling through an aggregate feed

**Review by EricBAndrews — CHANGES_REQUESTED** [review]
Pending above comment

**Comment by boscojwho** [comment]
> What do we think about enabling this only within a community? It works really nicely when the pinned posts are all collected at the top but it's a little weird to come across a random compact post when scrolling through an aggregate feed

Oh totally yes, didn't think about pinned posts showing up in aggregate feed, will look into it.

**Comment by boscojwho** [comment]
> What do we think about enabling this only within a community? It works really nicely when the pinned posts are all collected at the top but it's a little weird to come across a random compact post when scrolling through an aggregate feed

Added `guard let community, postTracker?.feedType == .community(community)` to only render pinned posts as compact in community feeds.

File: Mlem/Views/Shared/Posts/Feed Post.swift [hunk 1]
```diff
@@ -149,16 +149,30 @@ struct FeedPost: View {
             with: ConcreteEditorModel(post: postModel, operation: PostOperation.replyToPost)
         )
     }
+    
+    /// Render read pinned posts in less "in-your-face" way.
+    private var renderPinnedAsCompact: Bool {
+        /// Only render pinned posts in compact size in Community feed, ignore this behaviour in other feed types (e.g. Aggregate). [2024.01]
+        guard let community, postTracker?.feedType == .community(community) else {
```
Comment (EricBAndrews) [inline Mlem/Views/Shared/Posts/Feed Post.swift:156]: Something's funky with this logic--it's not consistently working when I test locally. This should do the trick:
```suggestion
        guard case .community = postTracker?.feedType else {
```

**Review by EricBAndrews — CHANGES_REQUESTED** [review]

**Comment by EricBAndrews** [comment]
Stale as of #1046

**Pull request closed without merging**